hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/TransferWithCustomRoyaltyFees.java

```diff
@@ -0,0 +1,426 @@
+    private static final long carolFee = 300L;
+
+    public static void main(String... args) {
+        new TransferWithCustomRoyaltyFees().runSuiteAsync();
+    }
+
+    @Override
+    public List<HapiSpec> getSpecsInSuite() {
+        return List.of(
+                transferNonFungibleWithRoyaltyHbarFee(),
+                transferNonFungibleWithRoyaltyFungibleFee(),
+                transferNonFungibleWithRoyaltyFallbackHbarFee(),
+                transferNonFungibleWithRoyaltyFallbackFungibleFee(),
+                transferNonFungibleWithRoyaltyHbarFeeInsufficientBalance(),
+                transferNonFungibleWithRoyaltyFungibleFeeInsufficientBalance(),
+                transferNonFungibleWithRoyaltyFallbackHbarFeeInsufficientBalance(),
+                transferNonFungibleWithRoyaltyFallbackFungibleFeeInsufficientBalance(),
+                transferNonFungibleWithRoyaltyFallbackFungibleFeeNoAssociation(),
+                transferNonFungibleWithRoyaltyFungibleFeeNoAssociation());
+    }
+
+    @HapiTest
+    public HapiSpec transferNonFungibleWithRoyaltyHbarFee() {
+        final var fourHundredHbars = 4 * ONE_HUNDRED_HBARS;
+        final var twoHundredHbars = 2 * ONE_HUNDRED_HBARS;
+        return defaultHapiSpec("transferNonFungibleWithRoyaltyHbarFee")
+                .given(
+                        newKeyNamed(NFT_KEY),
+                        cryptoCreate(hbarCollector).balance(0L),
+                        cryptoCreate(tokenReceiver).balance(ONE_MILLION_HBARS),
+                        cryptoCreate(tokenTreasury),
+                        cryptoCreate(tokenOwner).balance(0L),
+                        tokenCreate(nonFungibleToken)
+                                .treasury(tokenTreasury)
+                                .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                                .initialSupply(0)
+                                .supplyKey(NFT_KEY)
+                                .supplyType(TokenSupplyType.INFINITE)
+                                .withCustom(royaltyFeeNoFallback(1, 2, hbarCollector)),
+                        tokenAssociate(tokenReceiver, nonFungibleToken),
+                        tokenAssociate(tokenOwner, nonFungibleToken),
+                        mintToken(nonFungibleToken, List.of(ByteStringUtils.wrapUnsafely("meta1".getBytes()))),
+                        cryptoTransfer(movingUnique(nonFungibleToken, 1L).between(tokenTreasury, tokenOwner)))
+                .when(cryptoTransfer(
+                        movingUnique(nonFungibleToken, 1L).between(tokenOwner, tokenReceiver),
+                        movingHbar(fourHundredHbars).between(tokenReceiver, tokenOwner)))
+                .then(
+                        getAccountBalance(tokenOwner)
+                                .hasTinyBars(twoHundredHbars)
+                                .hasTokenBalance(nonFungibleToken, 0),
+                        getAccountBalance(tokenReceiver).hasTokenBalance(nonFungibleToken, 1),
```

Comment:
We can also validate that `tokenReceiver` has 400 hbars less compared to the balance before the `cryptoTransfer`.